### PR TITLE
[TS migration] Migrate '[Components Group 1]' component to TypeScript

### PR DESCRIPTION
**Pull request by rayane-d**
### Fixed Issues

$ https://github.com/Expensify/App/issues/36273
$ https://github.com/Expensify/App/issues/36125
$ https://github.com/Expensify/App/issues/36126
$ https://github.com/Expensify/App/issues/36127
$ https://github.com/Expensify/App/issues/36128
PROPOSAL: N/A

### Tests

For `SAMLLoadingIndicator` (from https://github.com/Expensify/App/pull/29526):

1. Create an account with a validated domain on expensify.com
2. Configure your account to have SAML enabled but not required
3. Enter the email for your account in New Expensify
4. Choose Use Single Sign On option
5. Confirm you briefly see an interstitial page about launching your SSO provider's login portal
6. Verify that the interstitial page is displayed correctly.

For `FocusModeNotification` (from https://github.com/Expensify/App/pull/30664):

1. Sign into an account that has more than 30 chats total that is not currently in focus mode and has not yet seen the modal.
2. Verify that a modal is presented that tells you that you have been switched to focus mode.
3. Press the "Got it" button and verify the modal is dismissed
4. Sign into an account that has more than 30 chats total that is not currently in focus mode and has not yet seen the modal.
5. Verify that a modal is presented that tells you that you have been switched to focus mode.
6. Tap the "here" link and verify the modal closes and you are brought to the preference page where you can switch back if you want.

For `ShowMoreButton`:
1. Click on FAB > Request Money > Manual
2. Enter an amount and click next
3. Choose an account
4. On the money confirmation page, verify that the "Show more" button is displayed correctly

For `TaxPicker` (From https://github.com/Expensify/App/pull/32550):

1. Make sure Tax is enabled in Old Dot.
2. Add some tax policies in Old Dot.
3. Add a member to your workspace.
4. select a user from your workspace with enabled policies.
5. Click on request money as either manual or scan.
6. enter amount.
7. click on Show More and select tax rate.
8. select a tax rate

- [x] Verify that no errors appear in the JS console

### Offline tests

N/A
### QA Steps

Same as tests
- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

https://github.com/Expensify/App/assets/77965000/6f6413e1-6b44-468c-8898-4257c028d9c6

android 2 Screenshot 2024-02-25 221545

https://github.com/Expensify/App/assets/77965000/1bee31ae-c809-49b4-838a-5bed3c4b0ed9

</details>

<details>
<summary>Android: mWeb Chrome</summary>

[mweb chrome 1.webm](https://github.com/Expensify/App/assets/77965000/805461ed-e1db-4a21-987b-65c098a67fdd)

mweb chrome 2 Screenshot 2024-02-25 203502

https://github.com/Expensify/App/assets/77965000/6bfe365b-0849-4f54-bc46-71f97a31c1c1

</details>

<details>
<summary>iOS: Native</summary>

https://github.com/Expensify/App/assets/77965000/98f3b0fb-f79d-4ea4-81fa-af8516ec468c

ios 2 Screenshot 2024-02-25 232914

</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/Expensify/App/assets/77965000/18949e3f-7670-452a-9c7e-e7b8e2acd148

mweb safari 2 Screenshot 2024-02-25 230629

https://github.com/Expensify/App/assets/77965000/a302a839-563e-482b-9ba5-837417bf7ba0

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/Expensify/App/assets/77965000/762acf8e-7544-4311-9018-7a3ff0def3d5

web 2 Screenshot 2024-02-25 202443

https://github.com/Expensify/App/assets/77965000/1ec394f9-48c5-49cb-bc4f-2a5e3d54faf3

</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/Expensify/App/assets/77965000/14c545c1-9b37-4ac9-9e29-6698090e46fe

desktop 2 Screenshot 2024-02-25 205313

https://github.com/Expensify/App/assets/77965000/20f55365-59c0-4ea9-9362-b546a3124545

</details>

### CODE REVIEW

File: src/components/TaxPicker.tsx [hunk 1]
```diff
@@ -1,16 +1,28 @@
...
 import * as OptionsListUtils from '@libs/OptionsListUtils';
 import * as TransactionUtils from '@libs/TransactionUtils';
 import CONST from '@src/CONST';
-import {defaultProps, propTypes} from './taxPickerPropTypes';
+import OptionsSelector from './OptionsSelector';
 
-function TaxPicker({selectedTaxRate, policyTaxRates, insets, onSubmit}) {
+type TaxPickerProps = {
+    /** Collection of tax rates attached to a policy */
+    policyTaxRates?: OptionsListUtils.PolicyTaxRateWithDefault;
```
Comment (s77rt) [inline src/components/TaxPicker.tsx:13]:
```suggestion
    policyTaxRates: PolicyTaxRateWithDefault;
```

Import type separately and make the prop required (then remove default value)

File: src/components/TaxPicker.tsx [hunk 2]
```diff
@@ -58,11 +70,11 @@ function TaxPicker({selectedTaxRate, policyTaxRates, insets, onSubmit}) {
         );
         return policyTaxRatesOptions;
     }, [policyTaxRates, searchValue, selectedOptions]);
-
-    const selectedOptionKey = lodashGet(_.filter(lodashGet(sections, '[0].data', []), (taxRate) => taxRate.searchText === selectedTaxRate)[0], 'keyForList');
+    const selectedOptionKey = sections?.[0]?.data?.filter((taxRate) => taxRate.searchText === selectedTaxRate)[0]?.keyForList;
```
Comment (s77rt) [inline src/components/TaxPicker.tsx:73]:
```suggestion
    const selectedOptionKey = sections?.[0]?.data?.find((taxRate) => taxRate.searchText === selectedTaxRate)?.keyForList;
```

File: src/components/TaxPicker.tsx [hunk 3]
```diff
@@ -58,11 +70,11 @@ function TaxPicker({selectedTaxRate, policyTaxRates, insets, onSubmit}) {
...
 
     return (
         <OptionsSelector
+            // @ts-expect-error TODO: TS migration
```
Comment (s77rt) [inline src/components/TaxPicker.tsx:77]:
```suggestion
            // @ts-expect-error TODO: Remove this once OptionsSelector (https://github.com/Expensify/App/issues/25125) is migrated to TypeScript.
```

**Review by s77rt — CHANGES_REQUESTED** [review]

**Comment by s77rt** [comment]
Please add the missing screenshots/videos

File: src/components/TaxPicker.tsx [hunk 4]
```diff
@@ -1,16 +1,29 @@
...
+    selectedTaxRate?: string;
+
+    /** Safe area insets */
+    insets?: EdgeInsets;
+
+    /** Callback to fire when a tax is pressed */
+    onSubmit: () => void;
+};
+
+function TaxPicker({selectedTaxRate = '', policyTaxRates = {} as OptionsListUtils.PolicyTaxRateWithDefault, insets, onSubmit}: TaxPickerProps) {
```
Comment (s77rt) [inline src/components/TaxPicker.tsx:26]:
```suggestion
function TaxPicker({selectedTaxRate = '', policyTaxRates, insets, onSubmit}: TaxPickerProps) {
```

**Review by s77rt — CHANGES_REQUESTED** [review]

**Comment by s77rt** [comment]
## Reviewer Checklist

- [ ] I have verified the author checklist is complete (all boxes are checked off).
- [ ] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the steps for local testing are in the `Tests` section
    - [ ] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] Android: Native
    - [ ] Android: mWeb Chrome
    - [ ] iOS: Native
    - [ ] iOS: mWeb Safari
    - [ ] MacOS: Chrome / Safari
    - [ ] MacOS: Desktop
- [ ] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [ ] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [ ] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [ ] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [ ] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [ ] If any new file was added I verified that:
    - [ ] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn't already exist
    - [ ] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG`)
- [ ] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [ ] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [ ] If the PR modifies the form input styles:
    - [ ] I verified that all the inputs inside a form are aligned with each other.
    - [ ] I added `Design` label so the design team can review the changes.
- [ ] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [ ] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>



https://github.com/Expensify/App/assets/16493223/5cdabcdf-6b34-4eab-964d-84fa2045406c


</details>

<details>
<summary>Android: mWeb Chrome</summary>



https://github.com/Expensify/App/assets/16493223/367bc07b-f1aa-4074-a06d-4f0ad2b570a8


</details>

<details>
<summary>iOS: Native</summary>



https://github.com/Expensify/App/assets/16493223/55437b13-6f24-4cae-8392-e53c61f0e450


</details>

<details>
<summary>iOS: mWeb Safari</summary>



https://github.com/Expensify/App/assets/16493223/3f891b21-31cf-43de-831d-227e6fae59fd


</details>

<details>
<summary>MacOS: Chrome / Safari</summary>





https://github.com/Expensify/App/assets/16493223/a2552d9f-84c1-42cc-9668-3f5a0665a6f0

https://github.com/Expensify/App/assets/16493223/0ef37f4d-158f-4c1f-be50-c115b178d526

</details>

<details>
<summary>MacOS: Desktop</summary>


desktop

</details>

**Comment by s77rt** [comment]
@rayane-djouah Please tag me once the screenshots / videos are added

**Comment by s77rt** [comment]
@rayane-djouah Any update here?

**Review by s77rt — APPROVED** [review]

**Review by s77rt — APPROVED** [review]

**Review by AndrewGable — DISMISSED** [review]
Looking good! Conflicts however

**Review by AndrewGable — APPROVED** [review]

**Pull request merged**